tools/cldr-apps/src/main/java/org/unicode/cldr/web/DiskDataCache.java

```diff
@@ -27,18 +52,57 @@ class DiskDataEntry {
         final CLDRFile diskFile;
         final Set<String> pathsForFile;
 
+        private final List<FixedCandidateProvider> fixedCandidateProviders = new LinkedList<>();
+
         public DiskDataEntry(CLDRLocale locale) {
             this.locale = locale;
             diskData = factory.makeSource(locale.getBaseName()).freeze();
             diskFile = factory.make(locale.getBaseName(), true).freeze();
             pathsForFile = getPathHeaderFactory().pathsForFile(diskFile);
+
+            addFixedCandidateProviders();
+        }
+
+        private void addFixedCandidateProviders() {
+            // Add all Candidate Providers here
+            fixedCandidateProviders.add(new GrammarCandidateProvider());
+            fixedCandidateProviders.addAll(personNameProviders);
+        }
+
+        /** Candidate provider for a regex */
+        class GrammarCandidateProvider extends PatternCacheCandidateProvider {
+            final GrammarInfo grammarInfo = sdi.getGrammarInfo(locale.getBaseName());
+
+            public GrammarCandidateProvider() {
+                super("^//ldml/units/unitLength.*/unit.*gender");
+            }
+
+            @Override
+            protected Collection<String> getCandidates() {
+                return grammarInfo.get(
+                        GrammaticalTarget.nominal,
+                        GrammaticalFeature.grammaticalGender,
+                        GrammaticalScope.units);
+            }
+        }
```

Comment:
this one is locale dependent depends on the supplementalDataInfo in the enclosing class